Bottom: 78d76d841f4c64ec0108d514c53c6c6798564362
Top:    3159dd6d8df231eeff5eec88274679b7afd5f91a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-02 13:30:20 +0000

hid/gtk: Call ghid_fill_pcb_polygon via (HID_DRAW *)->fill_pcb_polygon()

This indirection allows a more graceful fallback in the case where we
run out of stencil bits and replace our (HID_DRAW *)->fill_pcb_polygon()
vfunc pointer with the common_fill_pcb_polygon() implementation.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 43172c1..9124ba5 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -588,7 +588,7 @@ ghid_thindraw_pcb_polygon (hidGC gc, PolygonType *poly, const BoxType *clip_box)
 {
   common_thindraw_pcb_polygon (gc, poly, clip_box);
   ghid_set_alpha_mult (gc, 0.25);
-  ghid_fill_pcb_polygon (gc, poly, clip_box);
+  gui->graphics->fill_pcb_polygon (gc, poly, clip_box);
   ghid_set_alpha_mult (gc, 1.0);
 }
